File: 441-ArrangingCoins/441-ArrangingCoins.java
// Last updated: 9/14/2025, 9:14:03 AM
class Solution {
    public int arrangeCoins(int n) {
           long start = 0;
           long end = n;
           long res = 0;
           while(start <= end){
               long mid = start + (end - start)/2;
               long coins = mid * (mid + 1) / 2;
               if(coins > n){
                   end = mid - 1;
               } else{
                   start = mid + 1;
                   res = Math.max(mid , res);
               }
           }
           return (int)res;
    }
}

File: 441-ArrangingCoins/441-ArrangingCoins.py
# Last updated: 9/14/2025, 9:14:07 AM
class Solution:
    def arrangeCoins(self, n: int) -> int:
        start , end = 0 , n
        res = 0
        while(start <= end):
            mid = start + (end - start) // 2
            coins =  (mid / 2) * (mid + 1)   # n(n+1)/2
            if(coins > n):
                end = mid - 1
            else:
                start = mid + 1
                res = max(mid , res)
        return res